src/Features/Core/Portable/EditAndContinue/AbstractEditAndContinueAnalyzer.cs

```diff
@@ -617,9 +615,16 @@ private SyntaxNode FindStatement(SyntaxNode declarationBody, int position, out i
                         continue;
                     }
 
+                    // Finds a matching syntax node in the new source.
+                    // In case the node got deleted the newMember may be missing.
+                    // For those the active span should remain empty.
                     SyntaxNode newMember;
                     bool hasPartner = topMatch.TryGetNewNode(oldMember, out newMember);
-                    Debug.Assert(hasPartner);
+                    Debug.Assert(hasPartner || trackedSpan.IsEmpty);
```

Comment:
If a length check were used instead, would the change in ActiveStatementDescription.cs be unnecessary?